test/internal_rules.jl

```diff
@@ -136,282 +136,66 @@ end
-            fill!(dA.dval, 0.0)
-            fill!(db.dval, 0.0)
-            fill!(dx.dval, 0.0)
-            dx.dval[i] = 1.0
-            Enzyme.autodiff(
-                Reverse,
-                driver,
-                dx,
-                dA,
-                db
-            )
-            adJ[i, :] = db.dval
-        end
-        return adJ
-    end
-
-    function const_revJdxdb(driver, A, b)
-        adJ = zeros(length(b), length(b))
-        db = Duplicated(b, zeros(length(b)))
-        dx = Duplicated(zeros(length(b)), zeros(length(b)))
-        for i in 1:length(b)
-            copyto!(db.val, b)
-            fill!(db.dval, 0.0)
-            fill!(dx.dval, 0.0)
-            dx.dval[i] = 1.0
-            Enzyme.autodiff(
-                Reverse,
-                driver,
-                dx,
-                Const(A),
-                db
-            )
-            adJ[i, :] = db.dval
-        end
-        return adJ
-    end
-
-    function batchedrevJdxdb(driver, A, b)
-        n = length(b)
-        function seed(i)
-            x = zeros(n)
-            x[i] = 1.0
-            return x
-        end
-        adJ = zeros(size(A))
-        dA = BatchDuplicated(A, ntuple(i -> zeros(size(A)), n))
-        db = BatchDuplicated(b, ntuple(i -> zeros(length(b)), n))
-        dx = BatchDuplicated(zeros(length(b)), ntuple(i -> seed(i), n))
-            Enzyme.autodiff(
-                Reverse,
-                driver,
-                dx,
-                dA,
-                db
-            )
-        for i in 1:n
-            adJ[i, :] .= db.dval[i]
-        end
-        return adJ
-    end
-
-    function Jdxdb(driver, A, b)
-        x = A\b
-        dA = zeros(size(A))
-        db = zeros(length(b))
-        J = zeros(length(b), length(b))
-        for i in 1:length(b)
-            db[i] = 1.0
-            dx = A\db
-            db[i] = 0.0
-            J[i, :] = dx
+    @testset "cholesky" begin
+        @testset "with wrapper arguments" begin
+            @testset for Te in (Float64,), TS in (Symmetric, Hermitian), uplo in (:U, :L)
+                @testset for TA in (Const, Duplicated), Tret in (Const, Duplicated)
+                    _A = collect(exp(TS(rand(Te, 4, 4))))
+                    A = TS(_A, uplo)
+                    are_activities_compatible(Tret, TA) || continue
+                    test_forward(cholesky, Tret, (A, TA))
+                    test_reverse(cholesky, Tret, (A, TA))
+                end
+            end
         end
-        return J
-    end
-
-    function JdxdA(driver, A, b)
-        db = zeros(length(b))
-        J = zeros(length(b), length(b))
-        for i in 1:length(b)
-            db[i] = 1.0
-            dx = A\db
-            db[i] = 0.0
-            J[i, :] = dx
+        @testset "without wrapper arguments" begin
+            _square(A) = A * A'
+            @testset for Te in (Float64,)
+                @testset for TA in (Const, Duplicated), Tret in (Const, Duplicated)
```

Comment:
I think this is also missing the batch activities and complex eltypes (at least for `Matrix`)

Reply:
I think this is now addressed, but #1456 leads to test failures for complex element type.